### DISCUSSION TITLE
How can I modify the code to have a stronger print

### DISCUSSION THREAD

**Discussion by Tintouin** [NaitLee/Cat-Printer#89] [Q&A]
My print is too weak for my liking, and I wanted to have like a multi pass because the print is weaker on larger elements.

Imagine the process like this : FEED > PRINT > PRINT > PRINT > FEED > PRINT > PRINT > PRINT Etc...

I saw a similar issue that was opened a long time ago and @NaitLee said that we had to change the energy setting in a JSON file named _config_ but there are no config.json anymore.

If you don't want or can't create a whole command parameter or UI thing in the localhost, can you still give me some hints on how I can make this to be effective with both commandline tool and webUI tool. And moreover about how we can modify parts of your code to our liking ?

**Comment by NaitLee** [comment]
Sorry I missed this. Very busy that time.

The `config.json` is created by the web interface, and CLI doesn't need one (use `--energy` or `-e` as parameter, range from 0.0-1.0).

Example ("typewriter"): `python printer.py -e 0.5 -t 1,unifont,pf2 -`  
(unzip [this](https://github.com/NaitLee/Cat-Printer/files/8561846/unifont.pf2.zip) in cat-printer directory for that unifont)

0.5 could be very deep for most paper.

The printer machine can't preciously align paper (even if given same amount of feed & retract), so "multi-pass" can't really do that job.
